src/fast_array_utils/stats/_power.py

```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: MPL-2.0
+from __future__ import annotations
+
+from functools import singledispatch
+from typing import TYPE_CHECKING, cast
+
+from .. import types
+
+
+if TYPE_CHECKING:
+    from typing import Any, TypeVar
+
+    from numpy.typing import NDArray
+
+    # All supported array types except for disk ones and OutOfCoreDataset
+    Array = NDArray[Any] | types.CSBase | types.CupyArray | types.CupySparseMatrix | types.DaskArray
+
+    _Arr = TypeVar("_Arr", bound=Array)
+
+
+def power(x: _Arr, n: int, /) -> _Arr:
+    """Take array or matrix to a power."""
+    return _power(x, n)  # type: ignore[return-value]
```

Comment:
Why is this wrapper here?

Reply:
because of the stupid way `singledispatch`’s typing works (with the fallback function’s first argument having to be the supertype of all registered functions’ first arguments):

```
src/fast_array_utils/stats/_power.py:35: error: Dispatch type "Array" must be subtype of fallback function first argument "_Arr"  [misc]
```